docs/messaging/messaging-spans.md

```diff
@@ -261,6 +261,23 @@ batch of messages, or for no message at all (if it is signalled that no
 messages were received). For each message it accounts for, the "Process" or
 "Receive" span SHOULD link to the message's creation context.
 
+> [!IMPORTANT]
+> These conventions use spans links as the default mechanism to correlate
+> producers and consumer(s) because:
+>
+> - It is the only consistent trace structure that can be guaranteed,
+> given the many different messaging systems models available.
+>
+> - It is the only option to correlate produce and consumer(s) in batch scenarios
+> as a span can only have a single parent.
+>
+> - Offers a consistent experience for users analysing traces from different
+> messaging systems.
+
+Exclusively for single messages scenarios, "Process" or "Receive" span MAY
```

Comment:
```suggestion
Exclusively for single messages scenarios, "Process" span MAY
```

Receive spans can't use message context as a parent because they start before message is received and the context is known.

Reply:
Ah yeah of course!